indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskQueue.java

```diff
@@ -512,6 +516,8 @@ public boolean add(final Task task)
         SinglePhaseParallelIndexTaskRunner.DEFAULT_USE_LINEAGE_BASED_SEGMENT_ALLOCATION
     );
 
+    taskContextEnricher.enrichContext(task);
```

Comment:
Yes this is a very critical flow for druid. We cannot allow extensions to break this.
I would go one step further and add a future return type to the interface to . 
```
class TaskEnrichContext{
Future<Void>  enrichContext(Task task)
}
```
and then this code waits upto a max of 20 MS to proceed else timeout.

future.get(20MS)

Reply:
This is a normal interface (instead of an extension point), the default implementation is `Noop`.
Seems to me it will be overly protected by using try-catch and Future for such code.